src/app/@ansyn/menu-items/algorithms/effects/tasks.effects.ts

```diff
@@ -13,19 +18,22 @@ import {
 	LoadTasksFinishedAction,
 	RunTaskAction,
 	RunTaskFinishedAction,
-	SelectTaskAction,
+	SelectTaskAction, SetCurrentTaskStatus,
 	TasksActionTypes
 } from '../actions/tasks.actions';
 import { AlgorithmTask, AlgorithmTaskStatus } from '../models/tasks.model';
 import { TasksRemoteService } from '../services/tasks-remote.service';
+import { filter } from 'rxjs/internal/operators';
 
 @Injectable()
 export class TasksEffects {
 
 	@Effect()
 	loadTasks$: Observable<LoadTasksFinishedAction> = this.actions$.pipe(
 		ofType(TasksActionTypes.LOAD_TASKS),
-		withLatestFrom(this.store.select(selectTaskTotal), (action, total) => total),
+		withLatestFrom(this.store.select(selectAlgorithmTasksAreLoaded), (action, loaded) => loaded),
```

Comment:
1. why using twice `withLatestFrom` ?
2. Always better to use  `mergeMap`, `switchMap` will cancel old requests, we usually do not want to cancel http requests).

```typescript 
withLatestFrom(this.store.select(selectAlgorithmTasksAreLoaded), this.store.select(selectTaskTotal)),
filter(([action, loaded, total]) => !loaded),
mergeMap(([action, loaded, total]: [LoadTasksFinishedAction, boolean, number]) => {
....
```

Reply:
1. Using `withLatestFrom` twice can prevent redundant calls because of the filters. But I agree that it is better to avoid.
2. I disagree, in my short experience with rxjs, `mergeMap` is a trouble maker, causing unexpected activity. Is there a specific reason for using `mergeMap` _here_?